packages/react-select/src/components/LiveRegion.js

```diff
@@ -0,0 +1,197 @@
+      });
+    },
+    [ariaLiveMessages]
+  );
+
+  // Update aria live selected option when prop changes
+  useEffect(
+    function onChange() {
+      let message = '';
+      if (ariaSelection && messages.onChange) {
+        const { value, ...context } = ariaSelection;
+        message = messages.onChange(value, { ...context });
+      }
+      setAriaSelected(message);
+    },
+    [ariaSelection, messages]
+  );
+
+  // Formatting function for newly focused option/value
+  useEffect(
+    function onFocus() {
+      let focusMsg = '';
+      const focused = focusedOption || focusedValue;
+      const isSelected = !!(
+        focusedOption &&
+        selectValue &&
+        selectValue.includes(focusedOption)
+      );
+
+      if (focused && messages.onFocus) {
+        const context = {
+          label: getOptionLabel(focused),
+          isDisabled: isOptionDisabled(focused),
+          isSelected,
+          options,
+          type: focused === focusedOption ? 'option' : 'value',
+          value: selectValue,
+        };
+
+        focusMsg = messages.onFocus(focused, { ...context });
+      }
+      setAriaFocused(focusMsg);
+    },
+    [
+      focusedOption,
+      focusedValue,
+      getOptionLabel,
+      isOptionDisabled,
+      menuIsOpen,
+      messages,
+      options,
+      selectValue,
+    ]
+  );
+
+  useEffect(
+    function onFilter() {
+      let resultsMsg = '';
+      if (menuIsOpen && options.length && messages.onFilter) {
+        const resultsMessage = screenReaderStatus({
+          count: focusableOptions.length,
+        });
+        resultsMsg = messages.onFilter({ inputValue, resultsMessage });
+      }
+      setAriaResults(resultsMsg);
+    },
+    [
+      focusableOptions,
+      inputValue,
+      menuIsOpen,
+      messages,
+      options,
+      screenReaderStatus,
+    ]
+  );
+
+  useEffect(
+    function guidance() {
+      let guidanceMsg = '';
+      if (messages.guidance) {
+        const type = focusedValue ? 'value' : menuIsOpen ? 'menu' : 'input';
+        guidanceMsg = messages.guidance(type, {
+          isDisabled: focusedOption && isOptionDisabled(focusedOption),
+          isMulti,
+          isSearchable,
+          label: ariaLabel,
+          tabSelectsValue,
+        });
+      }
+      setAriaGuidance(guidanceMsg);
+    },
+    [
+      ariaLabel,
+      focusedOption,
+      focusedValue,
+      isDisabled,
+      isFocused,
+      isMulti,
+      isOptionDisabled,
+      isSearchable,
+      menuIsOpen,
+      messages,
+      tabSelectsValue,
+    ]
+  );
+
+  const ariaContext = `${ariaFocused} ${ariaResults} ${ariaGuidance}`;
+
+  return (
+    <A11yText
+      aria-live={ariaLive}
+      aria-atomic="false"
+      aria-relevant="additions text"
+    >
+      {isFocused && (
+        <React.Fragment>
+          <span id="aria-selection">{ariaSelected}</span>
+          <span id="aria-context">{ariaContext}</span>
+        </React.Fragment>
+      )}
```

Comment:
I've tested this, and find it troublesome after selection, blur, and regaining focus (Screenreader never again reveals the selected value because it hasn't changed).  Here's how I changed it locally to make it work, as a suggestion (let me know what you think):

(1) For these highlighted lines of code (removed condition and fragment):

```suggestion
<span id="aria-selection">{ariaSelected}</span>
<span id="aria-context">{ariaContext}</span>
```

(2) Made `ariaContext` conditional on `isFocused`:

```tsx
const ariaContext = isFocused ? `${ariaFocused} ${ariaResults} ${ariaGuidance}` : '';
```

(3) Made `ariaSelected` conditional on `isFocused`; in line 82, also adds `isFocused` to that `useEffect` dependencies:

```tsx
...
if (isFocused && ariaSelection && messages.onChange) {
...
```

And (4) added `removals` to `aria-relevant`

So now when the component loses focus the aria-live message becomes empty, sees that as relevant, and will announce the selected option again after announcing the label of the combobox.

Anyway, curious your thoughts on this.

Reply:
Which screenreader are you using for your testing out of curiosity? I am using VoiceOver on OSX and hadn't run into this though I may have to re-test after removing the "role=alert"

Reply:
@bozdoz the intention is to only read the "selected" option onChange. 

The issue you are referring to is something I thought about, but think it's a bit more complicated. It could be possible that the Select has a value or a defaultValue without any selection by the user. The other use case to be considered is if the select `isMulti`, then only the last selected value is read back but not all of values.

I think this needs more consideration overall, but my thoughts on this are to clear the ariaSelected value from the Select state when the select loses focus. This way we can derive and announce all selected values instead from the selectValue instead, though there is no existing support for announcing multiple options at present.

Reply:
> Which screenreader are you using for your testing out of curiosity? I am using VoiceOver on OSX and hadn't run into this though I may have to re-test after removing the "role=alert"

I am using the same. Though I did notice a change since updating to Big Sur.

Reply:
> The issue you are referring to is something I thought about, but think it's a bit more complicated. It could be possible that the Select has a value or a defaultValue without any selection by the user. The other use case to be considered is if the select `isMulti`, then only the last selected value is read back but not all of values.

For a defaultValue I would think you would still want the screen reader to announce that value; the user should know there's an option chosen and can move to another form input for example.  I haven't tested multiple choices yet, but thanks for the heads up.  That's something I wasn't thinking about.  Actually, I just tested a native <select multiple> element, and voiceover is truly awful with it.

Reply:
The selection event is currently only called by the `onChange` event so the ability to get the select's value would have to be derived differently. Will be addressed but wanted to maintain some parity to ensure nothing was affected especially sinceiIt's already a fairly big change stripping all of these state changes out of Select.js into its own component. 

By the way, one of the things I noticed while I was testing is that the screenreader's instructions about keyboard interactions in a listbox can interfere with the aria-live arrowing over different focused options because it's in the passive voice so it doesn't interrupt no matter how many different options the user interacts with. This is part of the reason I wanted to add aria-live as a prop to allow users to choose between off/passive/assertive to determine what's best for their own needs.

Reply:
I thought it best to remove the guidance message in my implementation.  Do you think assertive works better than passive?

Reply:
I personally think so, but apparently not everyone https://github.com/JedWatson/react-select/issues/3353

The described behavior above only happened when I had passive voice. Assertive or even a non-standard string would allow the narrator to be interrupted with new events, whereas passive did not.

Reply:
Wow. Yeah, assertive is way better than passive!  I'll do some tests for default value and multiple values.

Reply:
So after seeing how great `assertive` is (and noting it does have a downfall of cutting off the screen reader from reading the combobox label), I realized we *could* move the aria-live attributes to the child tags instead.

For your consideration:

```tsx
<A11yText>
  <span
    aria-live="polite"
    aria-atomic="false"
    aria-relevant="all"
    id="aria-selection"
  >
    {ariaSelected}
  </span>
  <span
    aria-live="assertive"
    aria-atomic="false"
    aria-relevant="additions text"
    id="aria-context"
  >
    {ariaContext}
  </span>
</A11yText>
```

Here the screenreader will read the combobox label, and not be cut off by repeating the selected value (because it's set to polite).  Once the user has engaged with the component and is selecting options an assertive message pops up, which I believe is ideal since it corresponds with the user interrupting the label announcement (and subsequent option choices) with their own actions.  Anyway, this may be opinionated, but I think that covers a lot of the bases I've been trying to cover.  Also note the difference in `aria-relevant`.  

Let me know what you think! 😄 

Let me ekno

Reply:
I considered this and played around with the different aria relevant and atomic settings. The one thing that would have to be considered in your example (that is fixed in this PR) is that the aria-live nodes are present onMount as this apparently creates some issues for some screen readers otherwise. 

I'm not sure which parts should be made assertive or passive. There are arguments to make selected and focused assertive, but perhaps onResults too? I was playing around with adding support for loading, but the timing of the loading seemed to vary the messaging (especially when in passive mode) and there's further conversations to be had about how persistent the navigation guidance instructions should be (probably not needed on every new addition)

For a first pass, I think exposing string formatting functions and a universal aria-live prop will perhaps give us a better idea how to improve or perhaps even expose this as a public component so each of the components of the aria context and selection can be customized.

Reply:
I have not yet looked into async loading.  I can't imagine needing that announced at all, especially if the total results count changes.  But I will do some research into it.

Reply:
Also found that the original ticket has something similar with two elements, one polite, and one assertive: https://github.com/JedWatson/react-select/pull/3550/files#diff-7064e1eb0f7889ae735e705d019ebfa160fbb668d7733f73ee6f107b8cf16fadR1844-R1849

Reply:
Yes, and I think this needs more thought to better understand how we want to structure all the components of the ariaContext and would be worth doing that when we have more resources available to know how this all works with JAWS and NVDA.

I'd like to wait until we get more testing available for all of the screen readers before experimenting and testing these and possibly other DOM changes for other aspects of ariaContext.